### ISSUE TITLE
Document how to manually install MySQL 5.7 on Trusty

### ISSUE THREAD

**Issue by cotsog** [travis-ci/docs-travis-ci-com#1605]
We receive lots of requests for MySQL 5.7 and it doesn't come pre-installed.

We should document the following "manual" recipe:

```yaml
addons:
  apt:
    sources:
      - mysql-5.7-trusty
    packages:
      - mysql-server
      - mysql-client
```

And here's the recipe to reset the root password:

```yaml
before_install:
  - sudo mysql -e "use mysql; update user set authentication_string=PASSWORD('new_password') where User='root'; update user set plugin='mysql_native_password';FLUSH PRIVILEGES;"
  - sudo service mysql restart
```

**Comment by anonymous6936** [comment]
"Document" the following recipe
readme.file
addons:
  apt:
    sources:
      - mysql-5.7-trusty
    packages:
      - mysql-server
      - mysql-client
And here's the recipe to reset the root password:

before_install:
  - sudo mysql -e "use mysql; update user set authentication_string=PASSWORD('new_password') where User='root'; update user set plugin='mysql_native_password';FLUSH PRIVILEGES;"
  - sudo service mysql restart

**Comment by reyaan-boltman** [comment]
Thanks. This worked for me.

**Comment by plaindocs** [comment]
Related to  #1236

**Comment by martinarrieta** [comment]
This recipe have a problem.

The default mysql version on `mysql-5.7-trusty` is 5.6. There are some differences on the mysql.user table that does not allow you to create users after upgrade to 5.7. 

```
root@vagrant-ubuntu-trusty-64:~# mysql -u root
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 2
Server version: 5.7.21 MySQL Community Server (GPL)

Copyright (c) 2000, 2018, Oracle and/or its affiliates. All rights reserved.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> CREATE USER 'testuser'@'localhost' IDENTIFIED BY 'simpletest';
ERROR 3009 (HY000): Column count of mysql.user is wrong. Expected 45, found 43. Created with MySQL 50633, now running 50721. Please use mysql_upgrade to fix this error.
``` 

In order to solve this (and it is also recommended to do it after upgrade every mysql update), as the message say you need to run the mysql_upgrade command.

```
# mysql_upgrade -u root
Checking if update is needed.
Checking server version.
Running queries to upgrade MySQL server.
Checking system database.
mysql.columns_priv                                 OK
mysql.db                                           OK
mysql.engine_cost                                  OK
mysql.event                                        OK
mysql.func                                         OK
mysql.general_log                                  OK
mysql.gtid_executed                                OK
mysql.help_category                                OK
mysql.help_keyword                                 OK
mysql.help_relation                                OK
mysql.help_topic                                   OK
mysql.innodb_index_stats                           OK
mysql.innodb_table_stats                           OK
mysql.ndb_binlog_index                             OK
mysql.plugin                                       OK
mysql.proc                                         OK
mysql.procs_priv                                   OK
mysql.proxies_priv                                 OK
mysql.server_cost                                  OK
mysql.servers                                      OK
mysql.slave_master_info                            OK
mysql.slave_relay_log_info                         OK
mysql.slave_worker_info                            OK
mysql.slow_log                                     OK
mysql.tables_priv                                  OK
mysql.time_zone                                    OK
mysql.time_zone_leap_second                        OK
mysql.time_zone_name                               OK
mysql.time_zone_transition                         OK
mysql.time_zone_transition_type                    OK
mysql.user                                         OK
Upgrading the sys schema.
Checking databases.
sys.sys_config                                     OK
Upgrade process completed successfully.
Checking if update is needed.
root@vagrant-ubuntu-trusty-64:~# mysql -u root
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 4
Server version: 5.7.21 MySQL Community Server (GPL)

Copyright (c) 2000, 2018, Oracle and/or its affiliates. All rights reserved.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> CREATE USER 'testuser'@'localhost' IDENTIFIED BY 'simpletest';
Query OK, 0 rows affected (0.00 sec)
```

TL; DR; add mysql_upgrade to `before_install`

```before_install:
  - mysql_upgrade -u root 
```

**Comment by upupming** [comment]
Hi, I found there is an error when I follow [the documentation](https://docs.travis-ci.com/user/database-setup/#mysql):
![image](https://user-images.githubusercontent.com/24741764/44619865-230b5d00-a8be-11e8-8ee2-b14794115a4e.png)

[This](https://travis-ci.com/upupming/HITMers-node-js-server/builds/82880029) is my build log. It seems we need `sudo mysql_upgrade -u root` instead `sudo mysql_upgrade`.

![image](https://user-images.githubusercontent.com/24741764/44619890-682f8f00-a8be-11e8-83f5-10bd2d0d6a27.png)
